File: backend/fleets/src/domain/port/fleet_repository.rs
use async_trait::async_trait;
use auto_impl::auto_impl;

use shared::domain::types::id::FleetId;
use shared::infra::database::error::DatabaseError;

use crate::domain::model::Fleet;

#[async_trait]
#[auto_impl(Arc)]
pub trait FleetRepository: Clone + Sync + Send {
    async fn find_by_id(&self, id: FleetId) -> Result<Option<Fleet>, DatabaseError>;
    async fn find_by_name(&self, fleet_name: String) -> Result<Option<Fleet>, DatabaseError>;
    async fn insert(&self, fleet_name: String) -> Result<FleetId, DatabaseError>;
}

File: backend/src/main/kotlin/com/logisty/core/domain/port/FleetRepository.kt
package com.logisty.core.domain.port

import com.logisty.core.domain.model.Fleet
import com.logisty.core.domain.model.values.FleetId
import com.logisty.core.domain.model.values.FleetName

interface FleetRepository {
    fun findById(fleetId: FleetId): Fleet?

    fun findByName(fleetName: FleetName): Fleet?

    fun createFleet(fleetName: FleetName): FleetId
}
